src/repl.coffee

```diff
@@ -7,7 +7,8 @@ CoffeeScript = require './coffeescript'
 
 replDefaults =
   prompt: 'coffee> ',
-  historyFile: path.join process.env.HOME, '.coffee_history' if process.env.HOME
+  historyFile: do (p = process.env.XDG_CACHE_HOME or process.env.HOME) ->
+    path.join p, '.coffee_history' if p
   historyMaxInputSize: 10240
```

Comment:
Fair point about `$PWD`. Are you sure that the REPL works if `historyFile` is undefined? I’m not sure that’s ever been tested. Like if you change the current code to:

```coffee
historyFile: path.join process.env.HOME, '.coffee_history' if process.env.SOME_VARIABLE_THAT_DOESNT_EXIST
```